### PR TITLE
chore(transformer): Don't parse method bodies when we don't need to

### LINKED ISSUE #1874: Do not parse function bodies in the transformer
We do not need to parse function bodies and `package:analyzer` now supports this -- update the transformer code to avoid the extra work.

Thanks!

### PR DESCRIPTION
**Pull request by jakemac53**
fixes https://github.com/angular/angular/issues/1874

### CODE REVIEW

File: modules/angular2/src/transform/reflection_remover/remove_reflection_capabilities.dart [hunk 1]
```diff
@@ -25,6 +25,7 @@ Future<String> removeReflectionCapabilities(AssetReader reader,
 
   var codegen = new Codegen(reflectionEntryPointPath, newEntryPointPaths);
   return new Rewriter(code, codegen,
-          mirrorMode: mirrorMode, writeStaticInit: writeStaticInit)
-      .rewrite(parseCompilationUnit(code, name: reflectionEntryPointPath));
+      mirrorMode: mirrorMode, writeStaticInit: writeStaticInit).rewrite(
+      parseCompilationUnit(code,
```
Comment (jakemac53) [inline modules/angular2/src/transform/reflection_remover/remove_reflection_capabilities.dart]: Filed https://github.com/dart-lang/dart_style/issues/255 about this formatting

File: modules/angular2/test/transform/bind_generator/all_tests.dart [hunk 2]
```diff
@@ -9,6 +9,8 @@ import '../common/read_file.dart';
 
 var formatter = new DartFormatter();
 
+main() => allTests();
```
Comment (kegluneq) [inline modules/angular2/test/transform/bind_generator/all_tests.dart]: Rather than aliasing, would you rename all of these `allTests` methods to `main`?

Comment (jakemac53) [inline modules/angular2/test/transform/bind_generator/all_tests.dart]: We also import this library from other files, where it makes more sense to call allTests() than main().

**Comment by jakemac53** [comment]
Abandoning this for now, looks like it breaks some of the benchmarks :(.

**Pull request closed without merging**